src/mongocxx/test/spec/monitoring.cpp

```diff
@@ -28,63 +28,39 @@ MONGOCXX_INLINE_NAMESPACE_BEGIN
 namespace spec {
 
 using namespace mongocxx;
+using bsoncxx::to_json;
 
 void apm_checker::compare(bsoncxx::array::view expectations,
                           bool allow_extra,
                           const test_util::match_visitor& match_visitor) {
-    using bsoncxx::to_json;
+    auto is_kill_cursor = [](bsoncxx::document::value v) {
+        return v.view()["command_started_event"]["command"]["killCursors"];
+    };
 
     auto events_iter = _events.begin();
+    auto events_end = (_skip_kill_cursors)
+                          ? std::remove_if(_events.begin(), _events.end(), is_kill_cursor)
+                          : _events.end();
+
     for (auto expectation : expectations) {
-        if (events_iter == _events.end()) {
-            fprintf(stderr,
-                    "more expectations than events captured\nExpectations:\n\n%s\n\nCaptured:\n\n",
-                    to_json(expectations).c_str());
-            this->print_all();
-            REQUIRE(false);
-        }
-        REQUIRE(events_iter != _events.end());
+        REQUIRE(events_iter != events_end);
 
         auto expected = expectation.get_document().view();
+        CAPTURE(to_json(*events_iter), expectation);
```

Comment:
`fprintf` printed on a different buffer from the `catch2` statements so output was often out of order.
This `CAPTURE` gives output like: 
```
mongo-cxx-driver/src/mongocxx/test/spec/monitoring.cpp:50: FAILED:
  to_json(*events_iter) := "{ "command_started_event" ...
  expectation := { ...
```
If the command below fails.